plugins/orchestrator/src/components/ExecuteWorkflowPage/ExecuteWorkflowPage.tsx

```diff
@@ -0,0 +1,165 @@
+import { OrchestratorSupportButton } from '../OrchestratorSupportButton/OrchestratorSupportButton';
+import { WorkflowDialog } from '../WorkflowDialog';
+import { EditorViewKind } from '../WorkflowEditor';
+import { TitleFieldTemplate } from './TitleFieldTemplate';
+
+const WrappedForm = withTheme(require('@rjsf/material-ui-v5').Theme);
+
+interface ExecuteWorkflowPageProps {
+  initialState?: Record<string, JsonValue>;
+}
+
+export const ExecuteWorkflowPage = (props: ExecuteWorkflowPageProps) => {
+  const orchestratorApi = useApi(orchestratorApiRef);
+  const { workflowId } = useRouteRefParams(executeWorkflowRouteRef);
+  const [loading, setLoading] = useState(false);
+  const [schemaResponse, setSchemaResponse] =
+    useState<WorkflowDataInputSchemaResponse>();
+  const [workflowDialogOpen, setWorkflowDialogOpen] = useState<boolean>(false);
+  const [formState, setFormState] = useState(props.initialState);
+
+  const navigate = useNavigate();
+  const instanceLink = useRouteRef(workflowInstanceRouteRef);
+
+  useEffect(() => {
+    setLoading(true);
+    orchestratorApi
+      .getWorkflowDataInputSchema(workflowId)
+      .then(response => {
+        setSchemaResponse(response);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, [orchestratorApi, workflowId]);
+
+  const onExecute = useCallback(async () => {
+    const parameters: Record<string, JsonValue> = {};
+    if (schemaResponse?.schema && formState) {
+      for (const key in formState) {
+        if (formState.hasOwnProperty(key)) {
+          const property = formState[key];
+          Object.assign(parameters, property);
+        }
+      }
+    }
+
+    setLoading(true);
+    const response = await orchestratorApi.executeWorkflow({
+      workflowId,
+      parameters,
+    });
+    setLoading(false);
+
+    navigate(instanceLink({ instanceId: response.id }));
+  }, [
+    formState,
+    instanceLink,
+    navigate,
+    orchestratorApi,
+    schemaResponse,
+    workflowId,
+  ]);
+
+  const onFormChanged = useCallback(
+    e => setFormState(current => ({ ...current, ...e.formData })),
+    [setFormState],
+  );
+
+  const executeButton = useMemo(
+    () => (
+      <Button variant="contained" color="primary" onClick={onExecute}>
+        Execute
+      </Button>
+    ),
+    [onExecute],
+  );
+
+  return (
+    <BaseOrchestratorPage>
+      <ContentHeader title="Execute">
+        <OrchestratorSupportButton />
+      </ContentHeader>
+      {loading && <Progress />}
+      {schemaResponse && (
+        <InfoCard
+          title={schemaResponse.workflowItem.definition.name ?? workflowId}
+          subheader={schemaResponse.workflowItem.definition.description}
+          action={
+            <>
+              <Button
+                variant="outlined"
+                color="secondary"
+                style={{ marginTop: 8, marginRight: 8 }}
+                onClick={_ => setWorkflowDialogOpen(true)}
```

Comment:
```suggestion
                onClick={() => setWorkflowDialogOpen(true)}
```

Reply:
There is a refactoring going on and this component will likely be replaced with new code.